plugin/metadata/provider.go

```diff
@@ -11,16 +11,27 @@ import (
 // Note: this method should work quickly, because it is called for every request
 // from the metadata plugin.
 type Provider interface {
-	// List of variables which are provided by current Provider. Must remain constant.
-	MetadataVarNames() []string
+	// List of variables which are provided by current Provider. Must not be changed during run-time.
+	VarInts() []string
+	VarStrings() []string
+
 	// Metadata is expected to return a value with metadata information by the key
 	// from 4th argument. Value can be later retrieved from context by any other plugin.
 	// If value is not available by some reason returned boolean value should be false.
-	Metadata(ctx context.Context, state request.Request, variable string) (interface{}, bool)
+	String(ctx context.Context, state request.Request, variable string) string
+	Int(ctx context.Context, state request.Request, variable string) int
```

Comment:
How will it work here when we find we would like a new type ? 
=> need to add 2 new functions to the interface Provider
Does it break the compatibility with all the already implemented plugin that will miss these 2 new interfaces ?
I guess yes ...

Would it be better, if we want to have typed metadata, to define enumeration of types and save the Name + Type + Value ?

And maybe the Metadata can offer the service to translate from Type to other Type of at least to storage (in the Getter/Setter)

Doing so, we keep only interface stable with only 2 functions:
- MetadataVars() => return array of (String, enumType)
- Metadata(ctx context.Context, state request.Request, variable string) (interface{}, bool) - here the type is already known.

and Metadata implement the convenient function for getting / setting the values. And these can be extended without changing the interface Provider.

What do you think ?